docs/02-app/01-building-your-application/01-routing/08-parallel-routes.mdx

```diff
@@ -280,7 +282,13 @@ If you want to navigate elsewhere and dismiss a modal, you can also use a catch-
   height="768"
 />
 
-```tsx filename="app/@authModal/[...catchAll]/page.js"
+```tsx filename="app/@authModal/[...catchAll]/page.tsx" switcher
+export default function CatchAll() {
+  return null
+}
+```
+
+```jsx filename="app/@authModal/[...catchAll]/page.jsx"switcher
 export default function CatchAll() {
   return null
 }
```

Comment:
Same issue here

Reply:
Same answer as above. Code block language switcher. When you switch languages, it will switch languages for all pages (it's saved in a cookie, I think). So if we don't supply both languages, even if the code is _exactly the same_, the code block will completely disappear if we don't list them both. 

Correct me if I am wrong here, @delbaoliveira.

Reply:
Yes, mostly. We don't save preferences to a cookie yet. The switching behavior only happens if there is both a sibling codeblock and the switcher prop in both. 

The main reason to add two code blocks even when the code is there same is so we have different file names, `../page.tsx` for Typescript folks and `../page.jsx` for JavaScript folks.